src/main/scala/analysis/MemoryModelMap.scala

```diff
@@ -49,19 +51,17 @@ class MemoryModelMap {
       val node = memoryRegions(exitNode).asInstanceOf[Set[Any]]
       // for each function exit node we get the memory region
       // and add it to the mapping
-      val stackRgns = node.collect{ case r: StackRegion => r }.toList.sortBy(_.start.asInstanceOf[BitVecLiteral].value)
-      val dataRgns = node.collect{ case r: DataRegion => r }.toList
-      val heapRgns = node.collect{ case r: HeapRegion => r }.toList
-      val accessRgns = node.collect{ case r: RegionAccess => r }.toList
+      val stackRgns = node.collect { case r: StackRegion => r }.toList.sortBy(_.start.asInstanceOf[BitVecLiteral].value)
```

Comment:
Why are we just assuming that StackRegion.start is a BitVecLiteral? If this is safe to do, shouldn't that be the type of it in the first place?

Reply:
@yousifpatti

Reply:
Yes it should be safe to assume so the type could change

Reply:
I think this is one that will make sense to come back to later, I'll leave it for now